components/containers/core/mojom/containers.mojom

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+module containers.mojom;
+
+// Represents a container that holds a separate browsing context/identity.
+// Containers allow users to maintain different identities within the same
+// browser profile, keeping cookies, storage, and other site data separate
+// between containers.
+struct Container {
+  // Unique identifier for the container.
+  string id;
+  // User-friendly name for the container.
+  string name;
+};
+
+// Factory interface to create handlers for the containers settings page. This
+// interface is implemented on the browser side and called by the renderer to
+// establish communication channels for container management.
+interface SettingsPageHandlerFactory {
+  // Creates a new settings page handler and establishes bidirectional
+  // communication between the browser and renderer processes.
+  CreateSettingsPageHandler(
+      pending_remote<SettingsPage> page,
+      pending_receiver<SettingsPageHandler> page_handler);
+};
+
+// Interface for renderer -> browser communication. This interface allows the
+// renderer (UI) to perform CRUD operations on containers, with all operations
+// being handled in the browser process.
+interface SettingsPageHandler {
```

Comment:
This might be better as `interface ContainersHandler` or `interface Containers` or `interface ContainersService` (usually only if it actually points to a profile or global service which implements it directly).